### PR TITLE
Code Insights: Fix popovers/context menus on the dashboard page

### PR DESCRIPTION
**Pull request by vovakulikov**
Fixes https://github.com/sourcegraph/sourcegraph/issues/42753
Fixes https://github.com/sourcegraph/sourcegraph/issues/43328
Fixes https://github.com/sourcegraph/sourcegraph/issues/43329
Fixes https://github.com/sourcegraph/sourcegraph/issues/43209

## Test plan
- Check that context menus are bottom-start aligned
- Check that context menus targets have active (menu open) visual states
- Check that the context menus list is left-aligned 
- Make sure that there is no visual overlap between the header and the active card's outline (see https://github.com/sourcegraph/sourcegraph/issues/43328)
- Check that the dashboard page doesn't have a broken scroll behaviour when one of the filters panel is open (see https://github.com/sourcegraph/sourcegraph/issues/43329)
- Check that the dashboard context menu three-dot icon in the dark theme has the right colour (see https://github.com/sourcegraph/sourcegraph/issues/43209)

## App preview:

- [Web](https://sg-web-vk-improve-insight-menu-ui.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (+0.02 kb) | 0.00% (+0.27 kb) | 0.00% (+0.25 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/vk/improve-insight-menu-ui/compare-report.html;tab=live_object) for a full comparison between the commits 5e81367 and f162226 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Comment by chwarwick** [comment]
@vovakulikov  With change to the filter panel I am now able to get into a state where I have multiple filter panels open.  https://www.loom.com/share/ca1a604abc304562a4c478ece5bffe6a

**Comment by vovakulikov** [comment]
@chwarwick I do believe we have this issue in the main at the moment and these PR changes don't change the way how we control the filter panel open state, I would like to fix this in a separate issue.

File: client/wildcard/src/components/Popover/components/popover-content/PopoverContent.tsx [hunk 1]
```diff
@@ -128,7 +129,7 @@ const FloatingPanelContent: FC<PropsWithChildren<FloatingPanelContentProps>> = p
     useEffect(
         () => () => {
             if (returnTargetFocus) {
-                targetElement?.focus()
```
Comment (chwarwick) [inline client/wildcard/src/components/Popover/components/popover-content/PopoverContent.tsx:131]: Will changing this default cause unexpected results in other places the popover is used?

**Review by chwarwick — APPROVED** [review]
Insights changes look good to me, unsure if there would be negative effects from the change of the popover in wildcard.

**Pull request merged**